src/Relation/HasOne.php

```diff
@@ -26,7 +26,7 @@ class HasOne extends AbstractRelation implements RelationInterface
      *
      * @param mixed $entity The native entity.
      *
-     * @return GenericEntity
+     * @return ?object
```

Comment:
Isn't this Entity itself ?

Reply:
At higher levels in the marshaled data structure, yes, but not necessarily at the lowest levels.